corebehrt/functional/utils.py

```diff
@@ -172,24 +165,111 @@ def truncate_patient(
     return truncated_patient
 
 
+def _get_non_priority_tokens(vocabulary: dict, low_priority_prefixes: List[str]) -> set:
+    """
+    Get tokens that start with low_priority_prefixes.
+    """
+    return {
+        v
+        for k, v in vocabulary.items()
+        if any(k.startswith(prefix) for prefix in low_priority_prefixes)
+    }
+
+
+def _drop_non_priority_tokens(
+    patient: pd.DataFrame,
+    non_priority_tokens: set,
+    truncation_length: int,
+    background_length: int,
+) -> pd.DataFrame:
+    """
+    Drop non-priority tokens from patient, keeping truncation_length - background_length tokens.
+    """
+    patient["non_priority"] = patient["concept"].isin(non_priority_tokens)
```

Comment:
I think it also modifies the patient df. Maybe just save the `non_priority` Series into a temporary var instead.

Reply:
Or, as `non_priority` seems to be used in the function below, maybe have a different function for computing it and adding it. Currently `_filter_invalid_positions` will fail if `_drop_non_priority_tokens` has not been called beforehand.